code/modules/antagonists/changeling/abomination.dm

```diff
@@ -0,0 +1,95 @@
+/*
+	- CHANGELING ABOMINATION -
+
+	Contents:
+		Reagent
+		Monster
+*/
+
+/datum/reagent/shiftium
+	name = "shifium"
+	description = "A changeling matter derived stimulant that overexcites the nervous system, forcing transformations in changelings."
+	color = "#9D5A99"
+	overdose_threshold = 8
+	taste_description = "savory with a bit of blood"
+	metabolization_rate = 0.3 * REAGENTS_METABOLISM
+	var/obj/shapeshift_holder/shapeshiftdata
+
+/datum/reagent/shiftium/on_mob_metabolize(mob/living/L)
+	L.adjustStaminaLoss(-10, 0)
+	..()
+
+/datum/reagent/shiftium/on_mob_end_metabolize(mob/living/L)
+	..()
+	if(!shapeshiftdata)
+		return
+	shapeshiftdata.restore()
+	REMOVE_TRAIT(L, CHANGELING_HIVEMIND_MUTE, type)
+
+/datum/reagent/shiftium/overdose_start(mob/living/L)
+	if (L.mind?.has_antag_datum(/datum/antagonist/changeling))
+		to_chat(L,"<span class='danger'>You struggle to maintain your form!</span>")
+	..()
+
+/datum/reagent/shiftium/overdose_process(mob/living/L)
+	L.adjustStaminaLoss(5, 0)
+	if(prob(volume * 2.5))
+		var/datum/antagonist/changeling/changeling = L.mind?.has_antag_datum(/datum/antagonist/changeling)
+		if(changeling)
+			metabolization_rate = 10 * REAGENTS_METABOLISM
```

Comment:
Jeez, this is going to make true changeling form really short-lived.

Reply:
Changeling True Form does not require the reagent to be inside the changeling's (original) body. It will be purged almost immediately after the transformation occurs.

Reply:
It turns out, shapeshifting stops original form's metabolism, rendering everything moot. Just make sure that it'll be purged quickly.

And also delete the whole `/datum/reagent/shiftium/on_mob_end_metabolize(mob/living/L)` which confounds matters further.

Reply:
Alright